### DISCUSSION TITLE
CVE-2024-3656 (Improper Privilege Management, 24.0.5)

### DISCUSSION THREAD

**Discussion by jj-semantic-web-com** [keycloak/keycloak#30904] [Q&A] [answered]
hi there, g'day! are there any details available on [CVE-2024-3656](https://security.snyk.io/vuln/SNYK-JAVA-ORGKEYCLOAK-7246770) ? i'd be especially interested in the attack vector and if this vulnerability can only be exploited under certain circumstances.

**Comment by sharpedavid** [comment]
I believe it's fixed by this commit:

https://github.com/keycloak/keycloak/commit/eedfd0ef51255f95d8b3f565f122a35a8de0f376

The affected methods appear to be:

UserResource.getUnmanagedAttributes()
ClientRegistrationPolicyResource.getProviders()

**Comment by codespearhead** [comment, marked as answer]
[CVE-2024-3656](https://security.snyk.io/vuln/SNYK-JAVA-ORGKEYCLOAK-7246770) was patched by this PR [1], and the patch is available in version `24.0.5` as per the [GitHub Advisory Database](https://github.com/advisories), which status was published on `2024-06-11T15:32:37Z`, a couple of weeks before this issue was created.

Snyk may have taken longer to update their vulnerability database, but it is now up to date.

[1] https://github.com/keycloak/keycloak/commit/d9f0c84b797525eac55914db5f81a8133ef5f9b1

@jj-semantic-web-com Can you [mark this comment as the answer](https://docs.github.com/en/discussions/managing-discussions-for-your-community/moderating-discussions#marking-a-comment-as-an-answer) so it doesn't show up as unanswered in the GitHub discussions tab?
